compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -809,18 +807,23 @@ public void visit(BLangImportPackage importPkgNode) {
         this.env.scope.define(pkgAlias, symbol);
     }
 
-    public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules, SymbolEnv env) {
+    public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
+                                       List<BLangCompilationUnit> compUnits, SymbolEnv env) {
         SymbolEnv prevEnv = this.env;
         this.env = env;
-        for (Name alias : predeclaredModules.keySet()) {
-            ScopeEntry entry = this.env.scope.lookup(alias);
-            if (entry == NOT_FOUND_ENTRY) {
-                this.env.scope.define(alias, predeclaredModules.get(alias));
-            } else {
-                while (entry.next != NOT_FOUND_ENTRY) {
-                    entry = entry.next;
+        for (BLangCompilationUnit compUnit : compUnits) {
+            for (Name alias : predeclaredModules.keySet()) {
+                BPackageSymbol symbol = duplicatePackagSymbol(predeclaredModules.get(alias));
+                symbol.compUnit = new Name(compUnit.name);
```

Comment:
We don't have to do this if the user has an import with this alias, right? Shall we move these two lines to L820?

Reply:
Let's try moving this out to a method and calling it only where necessary.

Reply:
Done.